lib/OpenQA/Git.pm

```diff
@@ -21,18 +22,19 @@ sub config ($self, $args = undef) {
     return $app->config->{'scm git'};
 }
 
-sub _prepare_git_command ($self, $args = undef) {
-    my $dir = $args->{dir} // $self->dir;
+sub _prepare_git_command ($self) {
+    my $dir = $self->dir;
```

Comment:
Maybe add a comment here as this function has some caveats and some things to improve.  for instance should it die or something if `$dir` is undef?

Reply:
I've moved the method right below `_run_cmd` and `_ssh_git_cmd`, in order to make it more obvious where it is called. Also added a check if `$dir` is set. I can also add a comment, do you have a suggestion ?

Reply:
Lets move on. looks good